File: leetsync/2358-number-of-ways-to-split-array/number-of-ways-to-split-array.go
func waysToSplitArray(nums []int) int {
	n := len(nums)
	result, totalSum, leftSum, rightSum := 0, 0, 0, 0
	for _, v := range nums {
		totalSum += v
	}
	for i := 0; i < n-1; i++ {
		leftSum += nums[i]
		rightSum = totalSum - leftSum
		if leftSum >= rightSum {
			result++
		}
	}
	return result
}

File: leetsync/2358-number-of-ways-to-split-array/number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        int n = nums.length;
        int result = 0;
        long totalSum = 0;
        long leftSum = 0, rightSum = 0;
        for (int num : nums) {
            totalSum += num;
        }
        for (int i = 0; i < n - 1; i++) {
            leftSum += nums[i];
            rightSum = totalSum - leftSum;
            if (leftSum >= rightSum) {
                result++;
            }
        }
        return result;
    }
}